prov/shm/src/smr_cntr.c

```diff
@@ -32,16 +32,30 @@
 
 #include "smr.h"
 
+static struct fi_ops_cntr smr_cntr_ops = {
+	.size = sizeof(struct fi_ops_cntr),
+	.read = ofi_cntr_read,
+	.readerr = ofi_cntr_readerr,
+	.add = ofi_cntr_add,
+	.adderr = ofi_cntr_adderr,
+	.set = ofi_cntr_set,
+	.seterr = ofi_cntr_seterr,
+	.wait = ofi_cntr_spin_wait
+};
```

Comment:
This uses only generic function.  Please move this to the common code.

Reply:
Do you think it would be possible to replace the wait function in the new util_cntr_no_wait_ops added by #5474 ? Or keep this as a separate util_cntr_spin_wait_ops?

Reply:
I was actually debating this in my head.  Technically, we could replace no wait calls with these spin wait versions.  But that could be misleading wrt what's expected by the API.  So, I like the idea of returning ENOSYS if the app didn't request waiting, but makes the call.

Reply:
Fair enough